### PR TITLE
OptimismPortal final proof setting

### PR DESCRIPTION
**Pull request by JuanCoRo**
This PR drafts the delivery of the following proofs to the Optimism monorepo:
- `test_finalizeWithdrawalTransaction_paused_reverts`
- `test_proveWithdrawalTransaction_paused_reverts`

**Before** merging, the following has to be accomplished
- [x] Write a comprehensive readme explaining the structure of the newly added `kontrol` folder structure
- [x] Understand if there are any performance hits that somehow arise from this re-structuring the repo

### CODE REVIEW

File: packages/contracts-bedrock/foundry.toml [hunk 1]
```diff
@@ -37,7 +37,7 @@ fs_permissions = [
   { access='read', path = './forge-artifacts/' },
   { access='write', path='./semver-lock.json' },
   { access='read-write', path='./.testdata/' },
-  { access='read', path='./kout' }
+  { access='read', path='./kout-deployment' }
```
Comment (spencerhaoxiao) [inline packages/contracts-bedrock/foundry.toml:40]: Please add the output directory into .gitignore

File: packages/contracts-bedrock/foundry.toml [hunk 2]
```diff
@@ -70,14 +70,14 @@ optimizer = false
...
 test = 'test/kontrol'
 script = 'scripts-kontrol'
 
-[profile.stategen]
-src = 'test/kontrol/state-change'
-out = 'state-out'
-test = 'test/kontrol/state-change'
+[profile.kprove]
+src = 'test/kontrol/proofs'
+out = 'kout-proofs'
```
Comment (spencerhaoxiao) [inline packages/contracts-bedrock/foundry.toml:81]: Also add the output directory to .gitignore

File: packages/contracts-bedrock/test/kontrol/kontrol/scripts/run-kontrol.sh [hunk 3]
```diff
@@ -155,9 +155,7 @@ use_booster=--use-booster
 #########################################
 tests=""
 #tests+="--match-test CounterTest.test_SetNumber "
-#tests+="--match-test StateDiffTest.setUp "
-tests+="--match-test StateDiffCheatcode.recreateDeployment "
-# tests+="--match-test StateDiffTest.testVerifyStateChange "
+tests+="--match-test OptimismPortalKontrol.test_proveWithdrawalTransaction_paused "
```
Comment (spencerhaoxiao) [inline packages/contracts-bedrock/test/kontrol/kontrol/scripts/run-kontrol.sh]: Please also add the test_finalizeWithdrawalTransaction_paused test here.

File: packages/contracts-bedrock/test/kontrol/proofs/OptimismPortal.k.sol [hunk 4]
```diff
@@ -1,11 +1,11 @@
 pragma solidity ^0.8.13;
 
-import { StateDiffCheatcode } from "./StateDiffCheatcode.sol";
-import { KontrolUtils } from "./KontrolUtils.sol";
+import { DeploymentSummary } from "./utils/DeploymentSummary.sol";
+import { KontrolUtils } from "./utils/KontrolUtils.sol";
 import { Types } from "src/libraries/Types.sol";
 import { OptimismPortalInterface as OptimismPortal, SuperchainConfigInterface as SuperchainConfig} from "./interfaces/KontrolInterfaces.sol";
```
Comment (spencerhaoxiao) [inline packages/contracts-bedrock/test/kontrol/proofs/OptimismPortal.k.sol]: Please run `forge fmt` for all the sol files in folder `test/kontrol`

File: packages/contracts-bedrock/test/kontrol/proofs/utils/DeploymentSummaryCode.sol [hunk 5]
```diff
@@ -2,7 +2,7 @@ pragma solidity ^0.8.13;
 
 import { Vm } from "forge-std/Vm.sol";
```
Comment (spencerhaoxiao) [inline packages/contracts-bedrock/test/kontrol/proofs/utils/DeploymentSummaryCode.sol]: This import is not used.

**Review by FWBlueWizard — COMMENTED** [review]
Looks good to me in run-kontrol.sh I would like to see if these tests pass. We could setup our internal repository to run everything in CI if it has not yet run for Optimism.

**Pull request merged**